### PR TITLE
Add s3 backend to help, fixes #4219

### LINKED ISSUE #4219: [gateway] minio gateway help doesn't show s3 backend.
## Expected Behavior
minio gateway shows s3 backend and how to run it.

## Current Behavior
minio gateway doesn't how s3 backend and how to run it.

## Possible Solution
Add help

## Steps to Reproduce (for bugs)

1. `minio gateway`

## Context
Manual testing

## Your Environment
All

### PR DESCRIPTION
**Pull request by nl5887**

### CODE REVIEW

File: cmd/gateway-main.go [hunk 1]
```diff
@@ -38,6 +38,7 @@ FLAGS:
   {{end}}{{end}}
 BACKEND:
   azure: Microsoft Azure Blob Storage. Default ENDPOINT is https://core.windows.net
+  s3: Amazon Simple Storage Service (S3).
```
Comment (harshavardhana) [inline cmd/gateway-main.go:42]: An example needs to be added as well - just mentioning this is not useful.

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Comment by krishnasrinivas** [comment]
you can add examples as well:
```
  2. Start minio gateway server for S3 backend.
      $ export MINIO_ACCESS_KEY=accesskey
      $ export MINIO_SECRET_KEY=secretkey
      $ minio gateway s3

  3. Start minio gateway server for S3 backend on custom endpoint.
      $ export MINIO_ACCESS_KEY=Q3AM3UQ867SPQQA43P2F
      $ export MINIO_SECRET_KEY=zuf+tfteSlswRu7BJ86wekitnifILbZam1KYY3TG
      $ minio gateway s3 https://play.minio.io:9000

```

**Comment by harshavardhana** [comment]
> 3. Start minio gateway server for S3 backend on custom endpoint.
>       $ export MINIO_ACCESS_KEY=Q3AM3UQ867SPQQA43P2F
>       $ export MINIO_SECRET_KEY=zuf+tfteSlswRu7BJ86wekitnifILbZam1KYY3TG
>       $ minio gateway s3 https://play.minio.io:9000

We should also show AWS S3.

**Comment by krishnasrinivas** [comment]
> We should also show AWS S3.

already there:

```
  2. Start minio gateway server for S3 backend.
      $ export MINIO_ACCESS_KEY=accesskey
      $ export MINIO_SECRET_KEY=secretkey
      $ minio gateway s3
```

**Comment by harshavardhana** [comment]
> already there:
> 
>   2. Start minio gateway server for S3 backend.
>       $ export MINIO_ACCESS_KEY=accesskey
>       $ export MINIO_SECRET_KEY=secretkey
>       $ minio gateway s3

Strange then why its not present in the master branch 

```
  {{.HelpName}} - {{.Usage}}

USAGE:
  {{.HelpName}} {{if .VisibleFlags}}[FLAGS]{{end}} BACKEND [ENDPOINT]
{{if .VisibleFlags}}
FLAGS:
  {{range .VisibleFlags}}{{.}}
  {{end}}{{end}}
BACKEND:
  azure: Microsoft Azure Blob Storage. Default ENDPOINT is https://core.windows.net

ENVIRONMENT VARIABLES:
  ACCESS:
     MINIO_ACCESS_KEY: Username or access key of your storage backend.
     MINIO_SECRET_KEY: Password or secret key of your storage backend.

EXAMPLES:
  1. Start minio gateway server for Azure Blob Storage backend.
      $ export MINIO_ACCESS_KEY=azureaccountname
      $ export MINIO_SECRET_KEY=azureaccountkey
      $ {{.HelpName}} azure
```

**Comment by harshavardhana** [comment]
I don't see it locally compiled from master branch. 

```
$ minio gateway
NAME:
  minio gateway - Start object storage gateway.

USAGE:
  minio gateway [FLAGS] BACKEND [ENDPOINT]

FLAGS:
  --address value  Bind to a specific ADDRESS:PORT, ADDRESS can be an IP or hostname. (default: ":9000")
  --quiet          Disable startup banner.
  --help, -h       Show help.
  
BACKEND:
  azure: Microsoft Azure Blob Storage. Default ENDPOINT is https://core.windows.net

ENVIRONMENT VARIABLES:
  ACCESS:
     MINIO_ACCESS_KEY: Username or access key of your storage backend.
     MINIO_SECRET_KEY: Password or secret key of your storage backend.

EXAMPLES:
  1. Start minio gateway server for Azure Blob Storage backend.
      $ export MINIO_ACCESS_KEY=azureaccountname
      $ export MINIO_SECRET_KEY=azureaccountkey
      $ minio gateway azure
```

**Comment by krishnasrinivas** [comment]
it's not on master, it's my comment. This was my comment:


----------------------------------------
you can add examples as well:
```
  2. Start minio gateway server for S3 backend.
      $ export MINIO_ACCESS_KEY=accesskey
      $ export MINIO_SECRET_KEY=secretkey
      $ minio gateway s3

  3. Start minio gateway server for S3 backend on custom endpoint.
      $ export MINIO_ACCESS_KEY=Q3AM3UQ867SPQQA43P2F
      $ export MINIO_SECRET_KEY=zuf+tfteSlswRu7BJ86wekitnifILbZam1KYY3TG
      $ minio gateway s3 https://play.minio.io:9000

```

**Comment by deekoder** [comment]
So what is needed further here?

**Review by harshavardhana — APPROVED** [review]

**Review by krishnasrinivas — APPROVED** [review]

**Pull request merged**